### PR TITLE
Replace mpi_cpp_format

### PR DESCRIPTION
**Pull request by luator**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description
Instead of our own implementation use the `run-clang-format` script from [here](https://github.com/Sarcasm/run-clang-format).

The main advantage is that it allows specifying files/directories that should be excluded from the formatting, either via `--exclude` or by adding a file `.clang-format-ignore` to the repository.

The copy of run-clang-format added here is extended a bit to allow specifying a _clang-format file.

The `cpp_format.py` module is removed and `mpi_cpp_format` is replaced by a thin wrapper around `run-clang-format` that sets some parameters such that the behaviour is mostly the same as with the old `mpi_cpp_format`.

One notable difference is that the script now does not print anything to stdout anymore.  Not sure if this is a problem?

## How I Tested
By running it locally.

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

**Comment by luator** [comment]
With `run-clang-format` it is also very easy to add a formatting check to pull requests on GitHub, see the [trifinger_object_detection package](https://github.com/open-dynamic-robot-initiative/trifinger_object_tracking/blob/cfcfecb7cc0432eb48184df61ad24e7b28cfc52b/.github/workflows/style_checker.yml#L14).

**Review by vincentberenz — APPROVED** [review]

File: python/mpi_cmake_modules/cpp_format.py [hunk 1]
```diff
@@ -1,130 +0,0 @@
-"""cpp_format.py
```
Comment (MaximilienNaveau) [inline python/mpi_cmake_modules/cpp_format.py:1]: why delete this file?
Just modify it.
This is also based on clang after all...

File: CMakeLists.txt [hunk 2]
```diff
@@ -68,6 +68,7 @@ install(DIRECTORY ${PROJECT_SOURCE_DIR}/cmake/
 # install the scripts.
 # cmake-format: off
 install(PROGRAMS
+  scripts/run-clang-format
```
Comment (MaximilienNaveau) [inline CMakeLists.txt:71]: clang is anyway used for C++ so why not call it cpp format in case another tool shows up?

**Review by MaximilienNaveau — CHANGES_REQUESTED** [review]
Dear @luator ,

Few remarks:
1/ the architecture of these executable is not exactly what you did. i.e. First you define you methods in the python module of the mpi_cmake_modules here:
    https://github.com/machines-in-motion/mpi_cmake_modules/tree/master/python/mpi_cmake_modules
2/ I would simply replace the cpp_format by you executable ans cpp_format is based on clang anyway you just improve the executable api here so no need for yet another executable.

**Comment by luator** [comment]
@MaximilienNaveau Since `run-clang-format` is a third-party script, I would like to keep changes on it at a minimum, so we can more easily update it, if needed.  Also keeping it as a self-contained executable script makes using it in GitHub workflows easier (see the example posted above).

However, I could probably move it to the modules as is and call its `__main__` function in `mpi_cpp_format`.  This would avoid installing more scripts to PATH and it should still be possible to call it directly via `python -m mpi_cmake_modules.run_clang_format` if needed (this can, for example, be useful to just check formatting without actually changing the files).  What do you think?

**Comment by jviereck** [comment]
I am not an expert on this. Maybe it's the best when I don't provide a review.

**Comment by MaximilienNaveau** [comment]
Ok I did not saw it was a third-party script,my bad...
I guess we leave it as is as an executable.
if we go for the `python3 -m mpi_cmake_module.run_clang_format`
we should then have:
- `python3 -m mpi_cmake_module.run_python_format`
- `python3 -m mpi_cmake_module.run_cpp_format`
- `python3 -m mpi_cmake_module.run_cmake_format`
- `python3 -m mpi_cmake_module.run_format # all together`
WDYT?

**Comment by luator** [comment]
Hm, I would rather not move the existing executables, to not break peoples workflows.

**Comment by luator** [comment]
@MaximilienNaveau Is there still something I should change?

**Pull request merged**